client.go

```diff
@@ -0,0 +1,323 @@
+}
+
+func (c *Rollbar) ErrorWithStackSkipWithExtras(level string, err error, skip int, extras map[string]interface{}) {
+	body := c.buildBody(level, err.Error(), extras)
+	data := body["data"].(map[string]interface{})
+	errBody, fingerprint := errorBody(err, skip)
+	data["body"] = errBody
+	data["fingerprint"] = fingerprint
+
+	c.push(body)
+}
+
+func (c *Rollbar) RequestErrorWithStackSkip(level string, r *http.Request, err error, skip int) {
+	c.RequestErrorWithStackSkipWithExtras(level, r, err, skip, noExtras)
+}
+
+func (c *Rollbar) RequestErrorWithStackSkipWithExtras(level string, r *http.Request, err error, skip int, extras map[string]interface{}) {
+	body := c.buildBody(level, err.Error(), extras)
+	data := body["data"].(map[string]interface{})
+
+	errBody, fingerprint := errorBody(err, skip)
+	data["body"] = errBody
+	data["fingerprint"] = fingerprint
+
+	data["request"] = c.errorRequest(r)
+
+	c.push(body)
+}
+
+// -- Message reporting
+
+func (c *Rollbar) Message(level string, msg string) {
+	c.MessageWithExtras(level, msg, noExtras)
+}
+
+func (c *Rollbar) MessageWithExtras(level string, msg string, extras map[string]interface{}) {
+	body := c.buildBody(level, msg, extras)
+	data := body["data"].(map[string]interface{})
+	data["body"] = messageBody(msg)
+
+	c.push(body)
+}
+
+// -- Misc.
+
+func (c *Rollbar) Wait() {
+	c.waitGroup.Wait()
+}
+
+// Build the main JSON structure that will be sent to Rollbar with the
+// appropriate metadata.
+func (c *Rollbar) buildBody(level, title string, extras map[string]interface{}) map[string]interface{} {
+	timestamp := time.Now().Unix()
+	data := map[string]interface{}{
+		"environment":  c.Environment,
+		"title":        title,
+		"level":        level,
+		"timestamp":    timestamp,
+		"platform":     runtime.GOOS,
+		"language":     "go",
+		"code_version": c.CodeVersion,
+		"server": map[string]interface{}{
+			"host": c.ServerHost,
+			"root": c.ServerRoot,
+		},
+		"notifier": map[string]interface{}{
+			"name":    NAME,
+			"version": VERSION,
+		},
+	}
+
+	for k, v := range extras {
+		data[k] = v
+	}
+
+	return map[string]interface{}{
+		"access_token": c.Token,
+		"data":         data,
+	}
+}
+
+// Extract error details from a Request to a format that Rollbar accepts.
+func (c *Rollbar) errorRequest(r *http.Request) map[string]interface{} {
+	cleanQuery := filterParams(c.FilterFields, r.URL.Query())
+
+	return map[string]interface{}{
+		"url":     r.URL.String(),
+		"method":  r.Method,
+		"headers": flattenValues(filterParams(c.FilterHeaders, r.Header)),
+
+		// GET params
+		"query_string": url.Values(cleanQuery).Encode(),
+		"GET":          flattenValues(cleanQuery),
+
+		// POST / PUT params
+		"POST": flattenValues(filterParams(c.FilterFields, r.Form)),
+	}
+}
+
+// filterParams filters sensitive information like passwords from being sent to
+// Rollbar.
+func filterParams(pattern *regexp.Regexp, values map[string][]string) map[string][]string {
```

Comment:
maybe, but they are currently only being used in `client.go`, so it seemed to make more sense to have them there. They are private anyway.

I'm fine with either. Is that a make-or-break?